horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -58,32 +64,403 @@ public class RunServiceTest extends BaseServiceTest {
+      Transformer scalarTransformer = createExampleTransformerWithJsonPath("scalarT", schema, scalarFunction, scalarPath);
+
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, arrayTransformer, scalarTransformer);
+
+      ObjectNode data = runWithValue(42.0d, schema);
+
+      uploadRun(data,test.name);
+
+      DataSet first = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+      DataSet second = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+      DataSet third = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+
+      assertNotNull(first);
+      assertTrue(first.data.isArray());
+      String target = postFunctionSchemaUri(schema);
+      validateScalarArray(first, target);
+      validateScalarArray(second, target);
+      validateScalarArray(third, target);
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationChoosingSchema(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+
+      Schema schemaA = createExampleSchema("Aba", "Aba", "Aba", false);
+      NamedJsonPath path = createExampleNamedJsonPath("value", "$.value");
+      Transformer transformerA = createExampleTransformerWithJsonPath("A", schemaA, "value => ({\"by\": \"A\"})", path);
+
+      Schema schemaB = createExampleSchema("Bcb", "Bcb", "Bcb", false);
+      Transformer transformerB = createExampleTransformerWithJsonPath("B", schemaB, "value => ({\"by\": \"B\"})");
+
+      Test test = createTest(createExampleTest(getTestName(info)));
+      addTransformer(test, transformerA, transformerB);
+
+      uploadRun(runWithValue(schemaB, 42), test.name);
+      DataSet dataset = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+      assertNotNull(dataset);
+      assertTrue(dataset.data.isArray());
+      assertEquals(1, dataset.data.size());
+      assertEquals("B", dataset.data.get(0).path("by").asText());
+
+      assertNull(dataSetQueue.poll(50, TimeUnit.MILLISECONDS));
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithoutMatchFirstLevel(TestInfo info) throws InterruptedException {
+      Schema schema = createExampleSchema("Aca", "Aca", "Aca", false);
+      testTransformationWithoutMatch(info, schema, runWithValue(schema, 42));
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithoutMatchSecondLevel(TestInfo info) throws InterruptedException {
+      Schema schema = createExampleSchema("B", "B", "B", false);
+      testTransformationWithoutMatch(info, schema, JsonNodeFactory.instance.objectNode().set("nested", runWithValue(schema, 42)));
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testMultiSchemaWithoutTransformer(TestInfo info) throws InterruptedException {
```

Comment:
Looks like this duplicates `testTransformationNestedSchemasWithoutTransformers`

Reply:
I added this based on an outdated request. I'll remove `testMultiSchemaWithoutTransformer`.